src/ln/channelmanager.rs

```diff
@@ -439,7 +439,7 @@ impl ChannelManager {
 						(res, chan_entry.get().get_their_node_id(), Some(chan_entry.remove_entry().1))
 					} else { (res, chan_entry.get().get_their_node_id(), None) }
 				},
-				hash_map::Entry::Vacant(_) => return Err(HandleError{err: "No such channel", action: None})
+				hash_map::Entry::Vacant(_) => return Err(APIError::APIMisuseError{err: "No such channel"})
```

Comment:
Shouldnt this also be ChannelUnavailable (after all, there are always going to be races on the client's side where a channel gets (force-)closed between when they decide to close it and when they call close_channel).